build_tools/packaging/promote_from_rc_to_final.py

```diff
@@ -0,0 +1,278 @@
+spec.loader.exec_module(change_wheel_version)
+
+assert hasattr(
+    change_wheel_version, "change_wheel_version"
+), "change_wheel_version module does not expose change_wheel_version function"
+
+
+def parse_arguments(argv):
+    parser = argparse.ArgumentParser(
+        description="""Promotes packages from release candidate to final release (e.g. 7.10.0rc1 --> 7.10.0).
+
+Promotion works for for wheels and .tar.gz.
+Wheels version is determined by python library to interact with the wheel.
+For tar.gz., the version is extract from <.tar.gz>/PKG-INFO file.
+"""
+    )
+    parser.add_argument(
+        "--input_dir",
+        help="Path to the directory that contains .whl and .tar.gz files to promote",
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        "--match-files",
+        help="Limits selection in '--input-dir' to files matchings this argument. Use wild cards if needed, e.g. '*rc2*' (default '*' to promote all files in '--input-dir')",
+        default="*",
+    )
+    parser.add_argument(
+        "--delete",
+        help="Deletes old file after successful promotion",
+        action="store_true",
+    )
+    return parser.parse_args(argv)
+
+
+def wheel_change_extra_files(new_dir_path, old_version, new_version):
+    # extract "rocm_sdk_core" from /tmp/tmp3swrl25j/wheel/rocm_sdk_core-7.10.0
+    package_name_no_version = (
+        str(new_dir_path).rsplit("/", 1)[-1].split(str(new_version))[0][:-1]
+    )
+
+    # correct capitalization
+    # of interest for amdgpu arch: wheels are all lower case
+    # (e.g.  rocm_sdk_libraries_gfx94x_dcgpu-7.10.0a20251028-py3-none-linux_x86_64.whl)
+    # but inside we have to match to rocm_sdk_libraries_gfx94X-dcgpu/ with a capital "X-dcgpu" instead of "x_dcgpu"
+    for dir in glob.glob(str(new_dir_path) + "/*"):
+        stripped_dir = str(dir).rsplit("/", 1)[-1]
+        if (
+            "gfx" in package_name_no_version
+            and "gfx" in stripped_dir
+            and len(package_name_no_version) == len(stripped_dir)
+        ):
+            package_name_no_version = stripped_dir
+            break
+
+    old_rocm_version = (
+        str(old_version)
+        if not "rocm" in str(old_version)
+        else str(old_version).split("+rocm")[-1]
+    )
+    new_rocm_version = (
+        str(new_version)
+        if not "rocm" in str(new_version)
+        else str(new_version).split("+rocm")[-1]
+    )
+
+    print("  Changing wheel-specific files that contain the version", end="")
+
+    if not "torch" in str(new_dir_path):  # rocm packages
+        files_to_change = [
+            f"{new_dir_path}/{package_name_no_version}/_dist_info.py",
+        ]
+    elif (
+        "torch" == package_name_no_version
+    ):  # only exactly torch and not triton, torchaudio, ..
+        print("torch")
+        files_to_change = [
+            f"{new_dir_path}/{package_name_no_version}/_rocm_init.py",
+            f"{new_dir_path}/{package_name_no_version}/version.py",
+        ]
```

Comment:
I'll test, but the logging/printing here may want to include more context and different formatting

(You could also share the output of running this script to make review easier)

Reply:
that print was some leftover from testing